vasppy/kpoints.py

```diff
@@ -41,3 +41,51 @@ def __init__(
             self.shift = np.array([0.0, 0.0, 0.0])
         else:
             self.shift = shift
+
+def get_subdivisions_from_kspacing(kspacing: float, reciprocal_lattice_vectors: np.ndarray) -> tuple[int, int, int]:
+    """
+    Calculate subdivisions along reciprocal lattice vectors from the miniumum allowed distance between k-points (KSPACING).
+
+    Args:
+        kspacing (float): The minimum allowed distance between k-points.
+        reciprocal_lattice_vectors (np.ndarray): The reciprocal lattice vectors. These can be retrieved from ASE as atoms.cell.reciprocal() or from Pymatgen as
+        structure.lattice.reciprocal_lattice_crystallographic.matrix.
+
+    Returns:
+        tuple[int, int, int]: The subdivisions along each reciprocal lattice vector.
+    """
+    subdivisions = []
+    for row in reciprocal_lattice_vectors:
+        magnitude = np.linalg.norm(row)
+        subdivision = np.max([1, np.ceil((magnitude * 2 * np.pi) / kspacing)])
+
+        subdivisions.append(int(subdivision))
+
+    return tuple(subdivisions)
```

Comment:
`mypy` is not able to determine the length of `subdivisions`, so complains about a type mismatch with the expected return value.

There are a few ways that probably work to get round this, with varying degrees of elegance.

I would probably try wrapping the conversion from vector to subdivision in a separate function (e.g., `vector_to_subdivision`, and then replacing the code above with e.g. 
```python
return tuple(map(vector_to_subdivision, reciprocal_lattice_vectors))
```